### PR TITLE
skip grant check when skip-grant-table enable

### PR DESCRIPTION
**Pull request by baurine**
Currently, if TiDB enables the `security.skip-grant-table`, running `show grants` and some other SQL statements will get errors, so we temporarily skip the grants check if `skip-grant-table` is enabled.

### CODE REVIEW

File: pkg/apiserver/user/verify_sql_user.go [hunk 1]
```diff
@@ -59,14 +60,20 @@ func VerifySQLUser(tidbClient *tidb.Client, userName, password string) (writeabl
 	if err != nil {
 		return false, err
 	}
-	// 2. Get grants
+	// 2. Check SkipGrantTable
+	// Note: Currently, if TiDB enable the skip-grant-table, running `show grants` will get error.
+	// So this is a workaround before the above bug is fixed.
+	if config.Security.SkipGrantTable {
+		return true, nil
```
Comment (baurine) [inline pkg/apiserver/user/verify_sql_user.go:67]: In this situation, should the user has the write privilege?

Comment (crazycs520) [reply pkg/apiserver/user/verify_sql_user.go:67]: I think the user can have the write privilege.

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Review by crazycs520 — APPROVED** [review]

**Comment by crazycs520** [comment]
@breeswish or @shhdgit PTAL

File: pkg/apiserver/user/verify_sql_user.go [hunk 2]
```diff
@@ -59,14 +60,20 @@ func VerifySQLUser(tidbClient *tidb.Client, userName, password string) (writeabl
 	if err != nil {
 		return false, err
 	}
-	// 2. Get grants
+	// 2. Check SkipGrantTable
+	// Note: Currently, if TiDB enable the skip-grant-table, running `show grants` will get error.
+	// So this is a workaround before the above bug is fixed.
```
Comment (shhdgit) [inline pkg/apiserver/user/verify_sql_user.go:65]: Does this mean that the non-root user has more privileges in this case? What is the impact on the tidb-dashboard?

Comment (crazycs520) [reply pkg/apiserver/user/verify_sql_user.go:65]: yes, if someone is connected to the tidb-server which `skip-grant-table=true`, he can do anything

**Review by shhdgit — APPROVED** [review]

**Pull request merged**